Automattic/jetpack
# Related Posts: allow author to set headline level

Fixes https://github.com/Automattic/jetpack/issues/34973

## Proposed changes:

The _Related Posts_ block can have a headline. Until now, this headline was an attribute that authors could edit in the settings panel. It was rendered inside an `h3` tag on the front end, regardless of the other page headings. This behavior could break the outline of the page and confuse some users, such as screen reader users.

This PR solves this issue by removing the headline from the settings and adding it as a child block instead. This way, the authors can choose the heading level and provide a consistent page outline to their site visitors.

| Before | After |
| - | - |
|<img width="283" alt="Screenshot 2024-01-31 at 10 55 42 AM" src="https://github.com/Automattic/jetpack/assets/1620183/07ac48b9-bf07-4bd2-a8eb-91fe3a9e59a9">|<img width="500" alt="Screenshot 2024-01-31 at 11 58 44 AM" src="https://github.com/Automattic/jetpack/assets/1620183/6bc5fa80-8039-4e64-ad04-782dcd06b47d">|


In more details, the PR:
- creates a deprecated version of the block to which current pages can fall back without breaking
- removes the `headline` and `displayHeadline` attributes
- adds a `core/heading` child block to the _Related Posts_ block
- ensures the block navigation has a consistent label

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






WARNING: before spinning up a site with this branch, check the _Prerequisites_ section.

### Prerequisites

- Spin up a test site that doesn't run this branch
- Create two posts: one with a _Related Posts_ block that has a heading and another one that has not
<img width="283" alt="Screenshot 2024-01-31 at 10 55 42 AM" src="https://github.com/Automattic/jetpack/assets/1620183/c6af8cdf-044a-448b-932e-3e747961e2b2">

- Keep these posts open in your browser tabs
- For each post, open the preview in a new tab

### Testing

- Switch the Jetpack version of your test site to this branch
- For each of the two posts and their preview
  - open the dev console
  - reload the page and check that everything looks okay
  - check there's no error in the console
- Create a new post
- Add a _Related Posts_ block
- Notice that you can add a heading from inside the block
<img width="500" alt="Screenshot 2024-01-31 at 11 58 44 AM" src="https://github.com/Automattic/jetpack/assets/1620183/59cdd4dd-7331-4953-a874-14e5fbf87d84">


- Enter a heading and choose a level
- Check it renders as expected on the front end
